File: Node Depths/allKindsOfNodeDepths.cpp
using namespace std;

class BinaryTree
{
public:
    int value;
    BinaryTree *left;
    BinaryTree *right;

    BinaryTree(int value)
    {
        this->value = value;
        left = NULL;
        right = NULL;
    }
};

int nodeDepths(BinaryTree *node, int depth = 0);

int nodeDepths(BinaryTree *node, int depth)
{
    if (node == NULL)
    {
        return 0;
    }

    return depth + nodeDepths(node->left, depth + 1) + nodeDepths(node->right, depth + 1);
}

File: Node Depths/allKindsOfNodeDepths.js
function nodeDepths(node, depth = 0) {
  if (node === null) {
    return 0;
  }

  return (
    depth + nodeDepths(node.left, depth + 1) + nodeDepths(node.right, depth + 1)
  );
}

class BinaryTree {
  constructor(value) {
    this.value = value;
    this.left = null;
    this.right = null;
  }
}

exports.allKindsOfNodeDepths = allKindsOfNodeDepths;
